.circleci/config.yml

```diff
@@ -134,35 +134,64 @@ jobs:
             echo "}" >> ~/project/decrypt.js
 
-      - run:
-          name: Decode secrets and environment variables
-          command: |
-            if [ -n "<< pipeline.parameters.SECRETS >>" ]; then
-              echo "Downloading and decoding secrets..."
+      - when:
+          condition:
+            not:
+              equal: ["upload", "<< pipeline.parameters.SOURCE_TYPE >>"]
+          steps:
+            - run:
+                name: Decode secrets and environment variables
+                command: |
+                  set -e  # Exit immediately if a command exits with a non-zero status
 
-              # Configure AWS CLI
-              aws configure set aws_access_key_id $AWS_ACCESS_KEY_ID
-              aws configure set aws_secret_access_key $AWS_SECRET_ACCESS_KEY
-              aws configure set default.region $AWS_REGION
+                  if [ -n "<< pipeline.parameters.SECRETS >>" ]; then
+                    echo "Downloading and decoding secrets..."
 
-              # Download secrets file from S3
-              aws s3 cp << pipeline.parameters.SECRETS >> ~/project/encrypted_secrets.json
+                    # Configure AWS CLI
+                    aws configure set aws_access_key_id $AWS_ACCESS_KEY_ID
+                    aws configure set aws_secret_access_key $AWS_SECRET_ACCESS_KEY
+                    aws configure set default.region $AWS_REGION
 
-              # Decrypt the secrets
-              DECRYPTED_JSON=$(node ~/project/decrypt.js "$(cat ~/project/encrypted_secrets.json)")
+                    # Download secrets file from S3
+                    aws s3 cp << pipeline.parameters.SECRETS >> ~/project/encrypted_secrets.json
 
-              echo "$DECRYPTED_JSON" | jq -r 'to_entries | map("export \(.key)=\(.value|@sh)") | .[]' > ~/project/decrypted_secrets.txt
-              source ~/project/decrypted_secrets.txt
+                    # Decrypt the secrets
+                    DECRYPTED_JSON=$(node ~/project/decrypt.js "$(cat ~/project/encrypted_secrets.json)")
 
-              # Clean up
-              rm ~/project/encrypted_secrets.json
-            fi
+                    echo "$DECRYPTED_JSON" | jq -r 'to_entries | map("export \(.key)=\(.value|@sh)") | .[]' > ~/project/decrypted_secrets.txt
+                    source ~/project/decrypted_secrets.txt
 
-            if [ -n "<< pipeline.parameters.ENVIRONMENT_VARIABLES >>" ]; then
-              echo "Processing environment variables..."
-              echo "<< pipeline.parameters.ENVIRONMENT_VARIABLES >>" | sed 's/^/export /' > ~/project/environment_variables.txt
-              source ~/project/environment_variables.txt
-            fi
+                    # Clean up
+                    rm ~/project/encrypted_secrets.json
+                  fi
+
+                  # Fetch environment variables from the One Click Server
+                  ENV_VARS_URL="<< pipeline.parameters.ONE_CLICK_SERVER_URL >>/projects/<< pipeline.parameters.PROJECT_ID >>/env-vars"
```

Comment:
https://github.com/earthfast/dashboard-runner/pull/31/files#diff-15f6621166019cfc732693a0c84c53275498d49506aa047bfa31489bdf0c99b4R54

what if PROJECT_ID is empty like it is here

Reply:
i've just added a check for that